src/block.js

```diff
@@ -2,12 +2,50 @@
  * External dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
+import classnames from 'classnames';
+/**
+ * Internal dependencies
+ */
+import { getColorClassName } from './util';
 
-export default function Block( { currentTotal, freeShippingFrom } ) {
+export default function Block( {
+	currentTotal,
+	freeShippingFrom,
+	textColor,
+	backgroundColor,
+	colorProps,
+} ) {
+	// Prepare progress width.
 	const progress = ( currentTotal / freeShippingFrom ) * 100;
-	const divWidth = ( progress > 100 ? 100 : progress ) + '%';
-	const divStyle = { width: divWidth };
+	const progressWidth = ( progress > 100 ? 100 : progress ) + '%';
+
+	// Prepare message color.
+	const messageColor =
+		textColor ?? colorProps?.style?.color ?? 'currentColor';
+	const messageClass = getColorClassName( 'color', messageColor );
+	const messageStyle = { color: colorProps?.style?.color ?? messageColor };
+
+	// Prepare border class and style.
+	const borderColor =
+		backgroundColor ?? colorProps?.style?.backgroundColor ?? 'currentColor';
+	const borderClass = getColorClassName( 'color', borderColor );
+	const borderStyle = {
+		color: colorProps?.style?.backgroundColor ?? borderColor,
+	};
+
+	// Prepare progress class and style.
+	const progressColor =
+		backgroundColor ?? colorProps?.style?.backgroundColor ?? 'currentColor';
+	const progressClass = getColorClassName( 'color', progressColor );
+	const progressStyle = {
+		width: progressWidth,
+		color: colorProps?.style?.backgroundColor ?? progressColor,
+	};
```

Comment:
Given that `Block` would never get `colorProps` and you're doing a lot of massaging to select either colorProps or attributes, why not move this logic to `Edit` and do the massaging there, ideally, `Block` wouldn't care from where the colors and classes are coming from, it should just apply style/class.

Reply:
> Given that Block would never get colorProps and you're doing a lot of massaging to select either colorProps or attributes, why not move this logic to Edit and do the massaging there, ideally, Block wouldn't care from where the colors and classes are coming from, it should just apply style/class.

I looked into this, but as this block is a dynamic block, the colors for the frontend need to be calculated on the fly. When I move the color logic into `Edit`, the frontend no longer receives any colors, unless I'm missing something. 

Do you want to run a quick test to see if you're able to move the color logic into `Edit` while ensuring that the colors remain visible in the frontend?

Reply:
> I looked into this, but as this block is a dynamic block, the colors for the frontend need to be calculated on the fly.

I don't see how would this makes a different? the inputs we use to calculate colors are just attributes, which are also available on the editor? The only different between editor and frontend is cart data, which is not included.

I can give this a try later this week, but I think you should be able to do it with some code massage, unless I'm missing another input that's only available in frontend?

Reply:
Thanks for your feedback, Nadir. I'll take another look to see how I can avoid the color calculation within the block and move that logic completely to `Edit`, so that `Block` only renders the colors.